internal/file/gcs.go

```diff
@@ -33,7 +33,11 @@ func gcsRequest(ctx context.Context, method, objectPath string) (*http.Response,
 		return nil, err
 	}
 
-	req, err := http.NewRequestWithContext(ctx, method, path.Join(gcsHTTPBaseURL, objectPath), nil)
+	u, _ := url.Parse(gcsHTTPBaseURL)
```

Comment:
I'd suggest panic'ing on URL parse error to catch any errors in gcsHTTPBaseURL over time.

```suggestion
	u, err := url.Parse(gcsHTTPBaseURL)
	if err != nil {
	  // This should never happen as base URL is defined by us.
	  panic("invalid GCS base URL: " + err.Error())
	}  
```